tests/functional_tests/iterator_tests/test_tuple.py

```diff
@@ -243,6 +271,34 @@ def test_field_of_tuple_input(fencil_processor):
         assert np.allclose(np.asarray(inp1) + np.asarray(inp2), out)
 
 
+def test_field_of_extra_dim_input(fencil_processor):
+    fencil_processor, validate = fencil_processor
+
+    shape = [5, 7, 9]
+    rng = np.random.default_rng()
+
+    inp1 = rng.normal(size=(shape[0], shape[1], shape[2]))
+    inp2 = rng.normal(size=(shape[0], shape[1], shape[2]))
+    inp = np.zeros(shape + [2])
+    for i in range(shape[0]):
+        for j in range(shape[1]):
+            for k in range(shape[2]):
+                inp[i, j, k, 0] = inp1[i, j, k]
+                inp[i, j, k, 1] = inp2[i, j, k]
```

Comment:
```suggestion
    inp = np.stack(inp1, inp2, axis=-1)
```

Reply:
thanks, I was to lazy to try to figure out to do that, but I assumed it exists. 🙃